src/main/java/com/stuypulse/robot/subsystems/Edwin.java

```diff
@@ -0,0 +1,229 @@
+package com.stuypulse.robot.subsystems;
+
+import com.kauailabs.navx.frc.AHRS;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.stuypulse.stuylib.control.Controller;
+import com.stuypulse.stuylib.control.feedback.PIDController;
+import com.stuypulse.stuylib.control.feedforward.Feedforward;
+import com.stuypulse.stuylib.math.Angle;
+
+import static com.stuypulse.robot.Constants.*;
+import static com.stuypulse.robot.Constants.EdwinSettings.*;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.SPI;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Edwin extends Robot {
+    
+    private final MotorControllerGroup left;
+    private final MotorControllerGroup right;
+    
+    private final DifferentialDrive drivetrain;
+
+    private final Encoder leftGreyhill;
+    private final Encoder rightGreyhill;
+
+    private final Controller leftController, rightController;
+    private double leftTargetSpeed, rightTargetSpeed;
+
+    private final DifferentialDriveOdometry odometry;
+    private final DifferentialDriveKinematics kinematics;
+    private final AHRS navx;
+
+    private final Field2d field;
+
+    public Edwin() {
+        left = new MotorControllerGroup(
+            new CANSparkMax(Ports.Edwin.LEFT_TOP, MotorType.kBrushless), 
+            new CANSparkMax(Ports.Edwin.LEFT_MIDDLE, MotorType.kBrushless),
+            new CANSparkMax(Ports.Edwin.LEFT_BOTTOM, MotorType.kBrushless)
+        );
+
+        right = new MotorControllerGroup(
+            new CANSparkMax(Ports.Edwin.RIGHT_TOP, MotorType.kBrushless), 
+            new CANSparkMax(Ports.Edwin.RIGHT_MIDDLE, MotorType.kBrushless),
+            new CANSparkMax(Ports.Edwin.RIGHT_BOTTOM, MotorType.kBrushless)
+        );
+
+        drivetrain = new DifferentialDrive(left, right);
+
+        leftGreyhill = new Encoder(Ports.Edwin.Encoders.LEFT_A, Ports.Edwin.Encoders.LEFT_B);
+        rightGreyhill = new Encoder(Ports.Edwin.Encoders.RIGHT_A, Ports.Edwin.Encoders.RIGHT_B);
+        
+        setGreyhillDistancePerPulse(EdwinSettings.Encoders.GREYHILL_DISTANCE_PER_PULSE);
```

Comment:
change this to a sparkmax set something per something. make sure to also set hte velocity conversion, which for spark maxes is DISTANCE_CONVERSION / 60.0. 

this is because DISTANCE_COVERSION is from rotations to distance and VELOCITY_CONERSION must convert Rotations per minute to distance / second 

(hence why i divide the conversionf actor by 60 to get it from rotations per mintiue to rotations per second, and then its multipying rotations to get distance)